### PR TITLE
Update Hetzner CSI to v2.1.0

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:
Update Hetzner CSI to v2.1.0.

**Which issue(s) this PR fixes**:
xref #2562 

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

I have manually tested this PR to ensure the backwards compatibility with v1.6.0.

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
Update Hetzner CSI from v1.6.0 to v2.1.0
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

File: pkg/templates/images/images.go [hunk 1]
```diff
@@ -302,7 +294,12 @@ func optionalResources() map[Resource]map[string]string {
 		HetznerCCM: {"*": "docker.io/hetznercloud/hcloud-cloud-controller-manager:v1.13.2"},
 
 		// Hetzner CSI
-		HetznerCSI: {"*": "docker.io/hetznercloud/hcloud-csi-driver:1.6.0"},
+		HetznerCSI:                   {"*": "docker.io/hetznercloud/hcloud-csi-driver:2.1.0"},
+		HetznerCSIAttacher:           {"*": "k8s.gcr.io/sig-storage/csi-attacher:v3.2.1"},
+		HetznerCSIResizer:            {"*": "k8s.gcr.io/sig-storage/csi-resizer:v1.2.0"},
+		HetznerCSIProvisioner:        {"*": "k8s.gcr.io/sig-storage/csi-provisioner:v2.2.2"},
+		HetznerCSILivenessProbe:      {"*": "k8s.gcr.io/sig-storage/livenessprobe:v2.3.0"},
+		HetznerCSINodeDriverRegistar: {"*": "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.2.0"},
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/templates/images/images.go]: You're right, updated the PR.

**Pull request merged**